lib/matplotlib/artist.py

```diff
@@ -1011,9 +1012,10 @@ def set_alpha(self, alpha):
                 f'alpha must be numeric or None, not {type(alpha)}')
         if alpha is not None and not (0 <= alpha <= 1):
             raise ValueError(f'alpha ({alpha}) is outside 0-1 range')
-        self._alpha = alpha
-        self.pchanged()
-        self.stale = True
+        if alpha != self._alpha:
+            self._alpha = alpha
+            self.pchanged()
+            self.stale = True
```

Comment:
For a follow-up PR I wonder if we should move `self.stale = True` into the `pchanged` definition? It seems like every call to `pchanged` is followed by `self.stale = True`...

Reply:
That can for sure be a good idea.